config/routes.rb

```diff
@@ -1,16 +1,33 @@
 Rails.application.routes.draw do
+  # scope :api, defaults: {format: :json} do
+  #   devise_for :users 
+  # end
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
+  namespace :api, defaults: {format: :json} do
+    resources :users, only: [:show] do
+      resources :posts, only: [:show] do
+        resources :comments, only: [:create]
```

Comment:
- [ ] In your API routes for the post you should have an index method, not a show method because your first requirement is that you have to show all the posts of a user :+1:
- [ ] Secondly, you should have an index method for the comments because the second requirement is that you should show all the comments of a post. I know you have done this in the posts controller but that's not the right place to do it :fearful: So kindly fix it :+1:
- [ ] After making the above changes kindly make changes in your posts and comments controller according to the above changes :+1:

Note: After making changes in your routes and controllers your changes will look something like the below in the image :+1:

## 1. Routes
![image](https://user-images.githubusercontent.com/93039359/216628994-5a9302bc-ede8-4eec-aa83-744f5139b2c5.png)
## 2. Posts Controller
![image](https://user-images.githubusercontent.com/93039359/216629217-fe6c99cb-9797-49bf-8489-e28ea81b9af1.png)
## 3. Comments Controller
![image](https://user-images.githubusercontent.com/93039359/216629571-918fb620-1874-4f7d-b03e-a23ede675493.png)